client/lib/user-settings/index.js

```diff
@@ -204,10 +221,11 @@ UserSettings.prototype.getSetting = function( settingName ) {
 	var setting = null;
 
 	// If we haven't fetched settings, or if the setting doesn't exist return null
-	if ( this.settings && 'undefined' !== typeof this.settings[ settingName ] ) {
-		setting = ( 'undefined' !== typeof this.unsavedSettings[ settingName ] )
-			? this.unsavedSettings[ settingName ]
-			: this.settings[ settingName ];
+	if ( this.settings && 'undefined' !== typeof get( this.settings, settingName ) ) {
+		setting =
+			'undefined' !== typeof get( this.unsavedSettings, settingName )
```

Comment:
I could be wrong here, but we may be able to use `this.isSettingUnsaved` here - if so that might help with readability :)